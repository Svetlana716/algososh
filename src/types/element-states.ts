export enum ElementStates {
  Default = "default",
  Changing = "changing",
  Modified = "modified",
};

export type SymbolType<T> = {
  data: T;
  state: ElementStates;
};

export type ExtraCircleType = {
  index: number;
  data: string | number | undefined;
  position: 'top' | 'bottom' | undefined;
};

export type SortingStep = {
  firstCheckableElement?: number;
  secondCheckableElement?: number;
  currentArray: number[];
  sortedElements: number[];
};