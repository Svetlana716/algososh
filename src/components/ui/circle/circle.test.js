import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";
import { screen, render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

describe('circle test case', () => {
    test ('circle with letters', () => {
        const tree = renderer
          .create(<Circle letter={'letter'} />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle without letters', () => {
        const tree = renderer
          .create(<Circle />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle with head', () => {
        const tree = renderer
          .create(<Circle letter={'letter'} head={'head'}/>)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle with circle in head', () => {
        const tree = renderer
          .create(<Circle letter={'letter'} head={<Circle letter={'letter'}/>}/>)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle with tail', () => {
        const tree = renderer
          .create(<Circle letter={'letter'} tail={'tail'}/>)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle with circle in tail', () => {
        const tree = renderer
          .create(<Circle letter={'letter'} tail={<Circle letter={'letter'}/>}/>)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('circle with index', () => {
        const tree = renderer
          .create(<Circle index={0} />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('small circle', () => {
        const tree = renderer
          .create(<Circle isSmall />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('default circle', () => {
        const tree = renderer
          .create(<Circle state={ElementStates.Default} />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('changing circle', () => {
        const tree = renderer
          .create(<Circle state={ElementStates.Changing} />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });

      test ('modified circle', () => {
        const tree = renderer
          .create(<Circle state={ElementStates.Modified} />)
          .toJSON();
    
          expect(tree).toMatchSnapshot();
      });
});