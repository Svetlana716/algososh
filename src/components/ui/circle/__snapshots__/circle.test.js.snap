// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle test case changing circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case circle with circle in head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-test-id="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          letter
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-test-id="tail"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case circle with circle in tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-test-id="circle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        >
          letter
        </p>
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
        data-test-id="tail"
      />
    </div>
  </div>
</div>
`;

exports[`circle test case circle with head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case circle with index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case circle with letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case circle with tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      letter
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-test-id="tail"
  >
    tail
  </div>
</div>
`;

exports[`circle test case circle without letters 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case default circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case modified circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;

exports[`circle test case small circle 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-test-id="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-test-id="tail"
  />
</div>
`;
